### PR TITLE
feat: export middleware from httpRouter entrypoint

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

Discord thread for context: https://discord.com/channels/687207715902193673/1301240156061433907

Exporting the middleware used by the default httpRouter implementation for custom implementations.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Yep sounds good to me! 👍 

Could you perhaps update https://backstage.io/docs/backend-system/core-services/http-router#configuring-the-service as well? Similar to the second half of https://backstage.io/docs/backend-system/core-services/root-config#configuring-the-service. I think it's fine to copy the entire existing implementation there.

**Comment by aramissennyeydd** [comment]
@Rugvip Yup, any ideas what's happening with the failing API report? I'm not able to replicate it locally 🤔

File: docs/backend-system/core-services/http-router.md [hunk 1]
```diff
@@ -70,4 +70,67 @@ access the incoming credentials.
...
+        httpAuth,
+        config,
+      });
+
+      router.use(createAuthIntegrationRouter({ auth }));
+      router.use(createLifecycleMiddleware({ lifecycle }));
+      router.use(credentialsBarrier.middleware);
+      router.use(createCookieAuthRefreshMiddleware({ auth, httpAuth }));
+
+      // Add a custom healthcheck endpoint for all plugins.
```
[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply docs/backend-system/core-services/http-router.md:120]: Yup, that's the only custom part - updated in [5170ecc](https://github.com/backstage/backstage/pull/27457/commits/5170ecc2485c4f198dfae5030f67b00f5ac93a8d)

**Comment by aramissennyeydd** [comment]
@Rugvip gentle bump - this should be ready for another round of 👀

**Review by Rugvip — APPROVED** [review]
Yep looks good, thank you! 👍 

Let's :shipit:

**Pull request merged**